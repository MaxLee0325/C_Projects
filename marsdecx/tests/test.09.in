12
0
1 0x08
1 0x0c
1 0x12
1 0x14
1 0x1b
1 0x1c
2 0x43 0xd8
3 0x62 0x69 0x00
4 0xca 0xf1 0x85 0x38
5 0xfb 0xff 0xff 0xff 0xf0
5 0xfc 0x00 0x00 0x00 0x10
