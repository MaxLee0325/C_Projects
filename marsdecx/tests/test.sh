#!/bin/sh
export LC_ALL=C

echo ======================================================
echo ====================== TEST $1 =======================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ______________Your_output______________ ____________Expected_Output_____________
   #echo +++++++++++++++++++++++++++++++++++++++ ++++++++++++++++++++++++++++++++++++++++
    diff -w -y -W 80 tests/test.$1.out tests/test.$1.expected
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
