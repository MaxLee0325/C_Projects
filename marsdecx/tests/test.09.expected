Line 2, integer 1: 0
Line 3, integer 1: -1
Line 4, integer 1: 1
Line 5, integer 1: -2
Line 6, integer 1: 3
Line 7, integer 1: -4
Line 8, integer 1: 123
Line 9, integer 1: 1234
Line 10, integer 1: 12345678
Line 11, integer 1: 1073741823
Line 12, integer 1: -1073741823
